flink/v2.1/flink/src/main/java/org/apache/iceberg/flink/sink/IcebergWriteAggregator.java

```diff
@@ -76,11 +85,17 @@ public void open() throws Exception {
 
   @Override
   public void finish() throws IOException {
-    prepareSnapshotPreBarrier(Long.MAX_VALUE);
+    prepareSnapshotPreBarrier(lastCheckpointId + 1);
   }
 
   @Override
   public void prepareSnapshotPreBarrier(long checkpointId) throws IOException {
+    if (checkpointId == lastCheckpointId) {
+      // Already flushed. This can happen when finish() above triggers flushing.
+      return;
+    }
+    this.lastCheckpointId = checkpointId;
```

Comment:
nit: newline

Reply:
>nit: newline

I initially had the newline, but I found it hurts readability. Line 97 is essentially an else statement, but wrapping inside an else block also looks verbose. I've added the newline.

>Also, should we have at least a log here, when we skip the precommit?

I've added a DEBUG log.